<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Curated tuberculosis analysis learning kit â€¢ curatedTBExplorer</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Curated tuberculosis analysis learning kit">
<meta name="description" content="curatedTBExplorer is an R package learning tool that lets users easily visualize, experiment with, and utilize machine learning to study curated tuberculosis data. It is to be used as an interactive R Shiny application to view and analyze the data without needing a high competence in R or Shiny. This can allow users that are interested in learning/studying about tuberculosis data take their first step.">
<meta property="og:description" content="curatedTBExplorer is an R package learning tool that lets users easily visualize, experiment with, and utilize machine learning to study curated tuberculosis data. It is to be used as an interactive R Shiny application to view and analyze the data without needing a high competence in R or Shiny. This can allow users that are interested in learning/studying about tuberculosis data take their first step.">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">curatedTBExplorer</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/wejlab/curatedTBExplorer/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="curatedtbexplorer">curatedTBExplorer<a class="anchor" aria-label="anchor" href="#curatedtbexplorer"></a>
</h1></div>
<p>CuratedTBData is a R data package that is accessible via Bioconductor and GitHub. This program facilitates integrative studies across TB gene expression datasets by improving the reliability and replicability of signature construction and validation. For research purposes, standardized gene expression measurements and clinical annotations are currently provided by the curated TBData. MultiassayExperiment objects, which effectively manage the raw data, several normalized versions of the data, metadata, and certain intermediate analysis findings, are where these data are maintained in R. We have reprocessed all 49 datasets and created standardized workflows for data preprocessing and annotation. In addition, the curatedTBData offers capabilities for subsetting and aggregating the transcriptome datasets with currently available tools, like ComBat and ComBat-Seq. In order to showcase the usefulness of the curatedTBData, we carried out an organized analysis and verification of several published TB gene signatures derived from blood. We investigate the prediction power of TB gene signatures in relation to distinguishing PTB from LTBI and healthy individuals (Control) in the 49 carefully selected studies. To train and assess gene signature ensembles, we also leverage dataset and datatype aggregation methods that are integrated into curatedTBData and are based on our prior research. Moreover, we demonstrate the use of an ensemble learning approach that combines the predictive power of several gene signatures.</p>
<div class="section level2">
<h2 id="before-installation">Before Installation<a class="anchor" aria-label="anchor" href="#before-installation"></a>
</h2>
<ul>
<li>Install BiocManager and related packages</li>
</ul>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"DESeq2"</span>, update <span class="op">=</span> <span class="cn">TRUE</span>, ask <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"GenomicRanges"</span>, update <span class="op">=</span> <span class="cn">TRUE</span>, ask <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"MultiAssayExperiment"</span>, update <span class="op">=</span> <span class="cn">TRUE</span>, ask <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"SummarizedExperiment"</span>, update <span class="op">=</span> <span class="cn">TRUE</span>, ask <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"TBSignatureProfiler"</span>, update <span class="op">=</span> <span class="cn">TRUE</span>, ask <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"curatedTBData"</span>, update <span class="op">=</span> <span class="cn">TRUE</span>, ask <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"scran"</span>, update <span class="op">=</span> <span class="cn">TRUE</span>, ask <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"SEtools"</span>, update <span class="op">=</span> <span class="cn">TRUE</span>, ask <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>The curatedTBExplorer requires R Version 4.1.</p>
<ul>
<li>Install the development version of the package from Github:</li>
</ul>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"devtools"</span>, quietly<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"wejlab/curatedTBExplorer"</span><span class="op">)</span></span></code></pre></div>
<p>If the installation fails, try:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>timeout<span class="op">=</span><span class="fl">400</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<h1>Content Summary:</h1>

<ol>
  <li><h3>Select/Download</h3></li>
  <p>The select and download page is where the user selects the datasets he wants to download and use.</p>
  
  <li><h3>Summarize</h3></li>
  <p>The summarize page is used for the filtering of data as well as the visualization of the data.</p>
  <p>When filtering data here, you will be removing certain samples based on chosen filter conditions.</p>
  
  <li><h3>TBSigProfiler</h3></li>
  <p>The TBSigProfiler page lets us view how effective pre-existing gene signatures are on your selected dataset.</p>
  <p>In addition, the user can add the gene signatures that they generate in the machine learning page and compare them to existing gene signatures.</p>

  <li><h3>Machine Learning</h3></li>
  <p>The machine learning page is where the user can generate gene signatures using machine learning algorithms on the data from selected studies.</p>
</ol>

<h1>Steps To Use Each Page</h1>
<h2>Select/Download</h2>

<ol>
  <li>Select the studies you want to use through the table and press the download button.</li>
  <img src="images/SelectDownloadPage-Downloading.png" alt="Image of the download page">

  <li>Once the download is complete, select the datasets you want to use out of the downloaded ones and press the confirm selected studies button.</li>
  <img src="images/SelectDownloadPage-Selecting.png" alt="Image of the selection of studies">

  <li>(Optional:) If there are covariate options for batch correction, you can press the confirm covariates button to perform batch correction on the selected data.</li>
  <img src="images/SelectDownloadPage-Covariates.png" alt="Image of the covariate selection for batch correction">

  <li>Now you can move on to the filter page or skip to the machine learning page</li>
</ol>
<br>
<h2>Summarize</h2>
<ol>
  <li>Select a covariate to filter by in the "Filter By" dropdown box. Then select a status in the next dropdown box. Ex. Filter By:Gender, Gender:Female.</li>
  <img>
  <p>For numberical covariates like age, it'll display a range instead of a dropdown box.</p>
  <img>

  <li>Once you are satisfied, press the "Add Filter" button. Keep in mind, you can add multiple filters before applying them</li>
  <img>

  <li>Press the "Apply Filter" button to apply your filters to your dataset</li>
  <p>The filter works by KEEPING the samples that have covariates with matching values. This means that if you apply a filter "Gender: Female", your data will only contain samples from females.</p>
  <img>

  <li>(Additional Info:) You can press the reset button to restore your dataset back to it's original form</li>
  <img>
  
  <li>Once you are satisfied with your data, you can move on to the TBSigProfiler page or skip to the machine learning page</li>
</ol>

<h2>TBSigProfiler</h2>
<ol>
  <li></li>
  <li></li>
  <li></li>
</ol>

<h2>Machine Learning</h2>
<ol>
  <li></li>
  <li></li>
  <li></li>
</ol>



  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/wejlab/curatedTBExplorer/" class="external-link">Browse source code</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing curatedTBExplorer</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Andrew Smith <br><small class="roles"> Author </small>  </li>
<li>Alex Sierra <br><small class="roles"> Author </small>  </li>
<li>Elie Wamana <br><small class="roles"> Author </small>  </li>
<li>Please Lukau <br><small class="roles"> Author </small>  </li>
<li>W. Evan Johnson <br><small class="roles"> Author, maintainer </small>  </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Andrew Smith, Alex Sierra, Elie Wamana, Please Lukau, W. Evan Johnson.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
